### DISCUSSION TITLE
Is there any way to ask GraphQLSchemaGenerator to write a local file in SDL after server start?

### DISCUSSION THREAD

**Discussion by RIP21** [ExpediaGroup/graphql-kotlin#977] [Q&A] [answered]
After I ran the app I want to write SDL that is logged into the console and can be retrieved from /sdl endpoint, into a local file.
Is there any way to do that? Maybe to auto-wire some schema printer bean (I'm, sadly, using Spring starter) and then call some method that returns such string once after the app started?

**Comment by smyrick** [comment, marked as answer]
@RIP21 The `GraphQLSchema` bean is created automatically and can be used by any part of your application

https://expediagroup.github.io/graphql-kotlin/docs/spring-server/spring-beans

So you can have a start up bean that depends on on the schema bean and prints the schema to a file.


```kotlin
fun printToFile(schema: GraphQLSchema) {
    File(fileName).writeText(schema.print())
}
```

**Reply by RIP21** [reply]
Awesome. Thanks!
